components/button/floating-buttons.js

```diff
@@ -0,0 +1,236 @@
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					-webkit-transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					-webkit-transform: translate(0,0);
+				}
+			}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this._container = null;
+		this._isRTL = false;
+		this._spacer = null;
+		this.minHeight = '500px';
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		window.removeEventListener('resize', this._reposition);
+		window.removeEventListener('scroll', this._reposition);
+	}
+
+	render() {
+		return html`
+			<lit-media-query
+  				.query="(max-height: ${this.minHeight})"
+  				@changed="${this._handleMediaQuery}">
+			</lit-media-query>
+			<div class="d2l-floating-buttons-container">
+				<div class="d2l-floating-buttons-inner-container">
+					<slot></slot>
+				</div>
+			</div>
+			<div class="d2l-floating-buttons-spacer"></div>
+		`;
+	}
+
+	firstUpdated() {
+		this._reposition = this._reposition.bind(this);
+		this.updateComplete.then(() => {
+
+			this._container = this.shadowRoot.querySelector('.d2l-floating-buttons-container');
+			this._spacer = this.shadowRoot.querySelector('.d2l-floating-buttons-spacer');
+
+			window.addEventListener('resize', this._reposition);
+			window.addEventListener('scroll', this._reposition);
+
+			this._isRTL = (getComputedStyle(this._container).direction === 'rtl');
+			this.updateComplete.then(() => {
+				this._reposition();
+				let prevDocumentHeight = document.body.offsetHeight;
+				setInterval(() => {
+					const documentHeight = document.body.offsetHeight;
+					if (prevDocumentHeight !== documentHeight) {
+						this._reposition();
+					}
+					prevDocumentHeight = documentHeight;
+				}, 100);
+			});
+		});
+	}
+
+	/**
+	 * Whether or not the buttons are floating.
+	 */
+	isFloating() {
+		return this._container.classList.contains('d2l-floating-buttons-floating');
+	}
+
+	_reposition() {
+		const containerRect = this._container.getBoundingClientRect();
+		this._spacer.style.height = `${containerRect.height}px`;
+
+		const spacerRect = this._spacer.getBoundingClientRect();
+
+		let containerTop;
+		const bodyScrollTop = document.body.scrollTop;
+		const isFloating = this._container.classList.contains('d2l-floating-buttons-floating');
+
+		if (isFloating) {
+			containerTop = spacerRect.top + bodyScrollTop;
+		} else {
+			containerTop = containerRect.top + bodyScrollTop;
+		}
+
+		const viewBottom = bodyScrollTop + window.innerHeight;
+		const innerContainer = this._container.querySelector('div');
+
+		if (!this.alwaysFloat &&
+			(this._viewportIsAtLeastMinHeight || ((containerTop + containerRect.height) <= viewBottom))) {
+
+			if (!isFloating) {
+				return;
+			}
+
+			this._container.classList.remove('d2l-floating-buttons-floating');
+			if (!this._isRTL) {
+				innerContainer.style.left = `${0}px`;
```

Comment:
With lit we don't actually want to modify style like this. See this comment from Dave B for more details https://github.com/BrightspaceUI/core/pull/18#issuecomment-475441395. Ideally here you can use the style map (see more-less e.g., https://github.com/BrightspaceUI/core/blob/master/components/more-less/more-less.js#L165 for example) or perhaps the class map.  You can likely move a lot of this `style` and `classList` modification into `render`. This PR https://github.com/BrightspaceUI/core/pull/19/files shows how I made these changes for `more-less`.

Reply:
Yeah it's a big paradigm shift for us, but ultimately ends up with much cleaner components that just happen to perform much better!